src/DssSpell.t.sol

```diff
@@ -354,33 +378,85 @@ contract DssSpellTest is DssSpellTestBase {
         // assertEq(dai.balanceOf(wallets.addr("DAIF_WALLET")), prevBalance + 67_863 * WAD);
     }
 
-    function testFlash() public {
+    function _setupRootDomain() internal {
+        vm.makePersistent(address(spell), address(spell.action()));
+
+        string memory root = string.concat(vm.projectRoot(), "/lib/dss-test");
+        config = ScriptTools.readInput(root, "integration");
+
+        rootDomain = new RootDomain(config, getRelativeChain("mainnet"));
+    }
+
+    function testL2OptimismSpell() public {
+        address l2TeleportGateway = BridgeLike(
+            chainLog.getAddress("OPTIMISM_TELEPORT_BRIDGE")
+        ).l2TeleportGateway();
+
+        _setupRootDomain();
+
+        optimismDomain = new OptimismDomain(config, getRelativeChain("optimism"), rootDomain);
+        optimismDomain.selectFork();
 
-        FlashAbstract flashLegacy = FlashAbstract(addr.addr("MCD_FLASH_LEGACY"));
-        FlashAbstract flashCurrent = FlashAbstract(addr.addr("MCD_FLASH"));
+        // Check that the L2 Optimism Spell is there and configured
+        L2Spell optimismSpell = L2Spell(0xC077Eb64285b40C86B40769e99Eb1E61d682a6B4);
 
-        assertEq(vat.wards(address(flashCurrent)), 1);
-        assertEq(vat.wards(address(flashLegacy)), 1);
+        L2Gateway optimismGateway = L2Gateway(optimismSpell.gateway());
+        assertEq(address(optimismGateway), l2TeleportGateway, "l2-optimism-wrong-gateway");
 
-        assertEq(flashLegacy.max(), 250 * MILLION * WAD);
-        assertEq(flashCurrent.max(), 250 * MILLION * WAD);
-        assertEq(flashLegacy.wards(pauseProxy), 1);
+        bytes32 optDstDomain = optimismSpell.dstDomain();
+        assertEq(optDstDomain, bytes32("ETH-GOER-A"), "l2-optimism-wrong-dst-domain");
+
+        // Validate pre-spell optimism state
+        assertEq(optimismGateway.validDomains(optDstDomain), 1, "l2-optimism-invalid-dst-domain");
+        // Cast the L1 Spell
+        rootDomain.selectFork();
 
         _vote(address(spell));
         _scheduleWaitAndCast(address(spell));
         assertTrue(spell.done());
 
-        assertEq(vat.wards(address(flashCurrent)), 1);
-        assertEq(vat.wards(address(flashLegacy)), 0);
+        // switch to Optimism domain and relay the spell from L1
+        // the `true` keeps us on Optimism rather than `rootDomain.selectFork()
+        optimismDomain.relayFromHost(true);
+
+        // Validate post-spell state
+        assertEq(optimismGateway.validDomains(optDstDomain), 0, "l2-optimism-invalid-dst-domain");
+    }
+
+    function testL2ArbitrumSpell() public {
```

Comment:
```suggestion
    function testL2ArbitrumSpell() public {
        // Ensure the Pause Proxy has some ETH for L2 Spells
        assertGt(pauseProxy.balance, 0);

```

Reply:
Right.
To go further - if we check the PP has a positive balance for Starknet we should check the Arbitrum gov relay has a positive balance as well.